androidApp/src/main/java/dev/johnoreilly/confetti/ui/ConfettiApp.kt

```diff
@@ -48,22 +58,48 @@ fun ConfettiApp(
         }
 
         fun onSwitchConference() {
-            appState.navigateToTopLevelDestination(ConferencesKey.route)
+            coroutineScope.launch {
+                confettiRepository.setConference(AppSettings.CONFERENCE_NOT_SET)
+                TopLevelDestination.values.forEach { destination ->
+                    // We *must* do this, else, the previously saved destinations will take
+                    // precedence over the new routes we pass in. This means that they will be
+                    // referencing the old conference argument, despite us sending in a new one,
+                    // which seems to be completely ignored.
+                    navController.clearBackStack(destination.routePattern)
+                }
```

Comment:
This section handles the backstack clearing point that I make in this PR's comment.

Reply:
This shouldn't be needed if the navOptions below are working correctly.